### DISCUSSION TITLE
Timed out fetching a new connection from the connection pool. More info: http://pris.ly/d/connection-pool (Current connection pool timeout: 10, connection limit: 20)

### DISCUSSION THREAD

**Discussion by andre0fastora** [prisma/prisma#22014] [Q&A]
### Question

I have a small web application that is using Prisma to hit a postgres DB hosted on Vercel. It was working fine yesterday but this morning I am getting the following error when hitting any of my api routes that require DB access: 

```
Timed out fetching a new connection from the connection pool. More info: http://pris.ly/d/connection-pool (Current connection pool timeout: 10, connection limit: 20)
    at vn.handleRequestError (/home/andre0fastora/projects/machine_tracker/node_modules/@prisma/client/runtime/library.js:123:7003)
    at vn.handleAndLogRequestError (/home/andre0fastora/projects/machine_tracker/node_modules/@prisma/client/runtime/library.js:123:6119)
    at vn.request (/home/andre0fastora/projects/machine_tracker/node_modules/@prisma/client/runtime/library.js:123:5839)
    at async l (/home/andre0fastora/projects/machine_tracker/node_modules/@prisma/client/runtime/library.js:128:9763)
    at async GET (webpack-internal:///(rsc)/./src/app/api/get/route.ts:10:22)
    at async eval (webpack-internal:///(rsc)/./node_modules/next/dist/server/future/route-modules/app-route/module.js:254:37) {
  clientVersion: '5.2.0',
  errorCode: undefined
}
```

### Information about Prisma Schema, Client Queries and Environment (optional)

// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url = env("POSTGRES_PRISMA_URL") // uses connection pooling
  directUrl = env("POSTGRES_URL_NON_POOLING") // uses a direct connection
}

model Machines {
  id String @id @default(uuid())
  number Int
  image String
  lastServiced DateTime
  binCount Int
  candy String
  moneyGatheredLastService Float
  moneyGatheredTotal Float
}
```

**Comment by ludralph** [comment]
Hi @andre0fastora 👋 

Thank you for raising this question.

Can you increase the connection pool timeout by setting the `pool_timeout` parameter to a value larger than the default (10 seconds) in your database connection URL. Here is an example:
```prisma
datasource db {
	provider = "postgresql"
	url = "postgresql://johndoe:mypassword@localhost:5432/mydb?connection_limit=5&pool_timeout=20"
}
```
In this example, the pool timeout is set to 20 seconds. Adjust this value according to your needs.

From the link that is part of the error message: https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/connection-pool#connection-pool-timeout

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
